packages/example/src/components/modals/SubtitlesModal.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect } from 'react';
+import { SpatialNavigationRoot } from '../../../../lib/src/spatial-navigation/components/Root';
+import { DefaultFocus } from '../../../../lib/src/spatial-navigation/context/DefaultFocusContext';
+import { useLockSpatialNavigation } from '../../../../lib/src/spatial-navigation/context/LockSpatialNavigationContext';
+import { Button } from '../../design-system/components/Button';
+import { Spacer } from '../../design-system/components/Spacer';
+import { GenericModal } from './GenericModal';
+import { useIsFirstRender } from '../../hooks/useIsFirstRender';
+
+interface SubtitlesModalProps {
+  isModalVisible: boolean;
+  setIsModalVisible: (isVisible: boolean) => void;
+  setSubtitles: (subtitles: string) => void;
+}
+
+export const SubtitlesModal = ({
+  isModalVisible,
+  setIsModalVisible,
+  setSubtitles,
+}: SubtitlesModalProps) => {
+  const { lock, unlock } = useLockSpatialNavigation();
+  const isFirstRender = useIsFirstRender();
+
+  // Locks the parent navigator when the modal is open and unlocks it when it's closed
+  useEffect(() => {
```

Comment:
there's a way better way to do this: 

```tsx
useEffect(() => {
  lock();
  return () => unlock();
}, [lock, unlock])
```

this will run when the component is mounted, everytime lock/unlock changes (and if it does, it runs both the previous unlock AND the next lock, so it's fine), and when it's unmounted (runs unlock)

Reply:
@pierpo can you let me know how to use in class base component in react native
 const { lock, unlock } = useLockSpatialNavigation();

Reply:
you can't use a hook in a class component, you should use a functional component 😉 

you can wait for this pr to be merged and take inspiration from the final code 👍